.github/workflows/coverage-report.yml

```
@@ -30,44 +30,51 @@ jobs:
             echo "is_dependabot=true" >> "$GITHUB_OUTPUT"
           else
             echo "is_dependabot=false" >> "$GITHUB_OUTPUT"
           fi
 
       - name: Check if the workflow should run
-        if: ${{ steps.dependabot-check.outputs.is_dependabot == 'false' }}
+        if: ${{ steps.dependabot-check.outputs.is_dependabot == 'false' }} && ${{ github.base_ref == 'refs/heads/main' }}
         run: echo "The workflow is allowed to run for this PR"
 
       - name: 🧲 Checkout
+        if: ${{ steps.dependabot-check.outputs.is_dependabot == 'false' }} && ${{ github.base_ref == 'refs/heads/main' }}
```

actionlint

if: condition "${{ steps.dependabot-check.outputs.is_dependabot == 'false' }} && ${{ github.base_ref == 'refs/heads/main' }}" is always evaluated to true because extra characters are around ${{ }}

(if-cond)